lib/faraday/adapter/http_gem.rb

```diff
@@ -21,10 +29,12 @@ def call(env)
               :headers => env[:request_headers],
               :proxy => proxy_options(env)
             }.merge(socket_options(env)))
-        rescue HTTP::ConnectionError => e
-          raise Faraday::Error::ConnectionFailed, e
-        rescue OpenSSL::SSL::SSLError => e
-          raise Faraday::SSLError, e
+        rescue *HTTP_EXCEPTIONS => err
+          if defined?(OpenSSL) && OpenSSL::SSL::SSLError === err
```

Comment:
I shamelessly took the exception handling pattern from the `Net::HTTP` adapter implementation https://github.com/lostisland/faraday/blob/master/lib/faraday/adapter/net_http.rb (reasoning it's one of the oldest and most mature) and `OpenSSL` isn't a part of the HTTP gem's gemspec so I guess there are scenarios when this would be an issue.